src/components/mint/module.cairo

```diff
@@ -200,11 +162,14 @@ mod Mint {
         }
 
         fn is_sold_out(self: @ContractState) -> bool {
-            self.get_available_value() == 0
-        }
-
-        fn is_canceled(self: @ContractState) -> bool {
-            self._mint_cancel.read()
+            let project_address = self._mint_carbonable_project_address.read();
+            let slot = self._mint_carbonable_project_slot.read();
+            let max_value = IAbsorberDispatcher { contract_address: project_address }
+                .get_project_value(slot);
+            let reserved_value = self._mint_reserved_value.read();
+            let total_value = IProjectDispatcher { contract_address: project_address }
+                .total_value(slot);
+            total_value + reserved_value >= max_value
```

Comment:
Short answer: It shouldn't happen if there is only one correct minter.

It might happen with _multiple_ minters, in that case we rely on the external view `total_value` and from the initializer requirements on `project_value`.